src/librustc_mir/build/expr/into.rs

```diff
@@ -146,11 +146,13 @@ impl<'a,'tcx> Builder<'a,'tcx> {
                 // start the loop
                 this.cfg.terminate(block, Terminator::Goto { target: loop_block });
 
-                this.in_loop_scope(loop_block, exit_block, |this| {
+                let might_break = this.in_loop_scope(loop_block, exit_block, move |this| {
                     // conduct the test, if necessary
                     let body_block;
-                    let opt_cond_expr = opt_cond_expr; // FIXME rustc bug
                     if let Some(cond_expr) = opt_cond_expr {
+                        // This loop has a condition, ergo its exit_block is reachable.
+                        this.find_loop_scope(expr_span, None).might_break = true;
```

Comment:
To me it sounds like it is better to keep state consistent in case somebody else wants to use it later for whatever reasons.